homeassistant/components/lutron/fan.py

```diff
@@ -0,0 +1,117 @@
+"""Lutron fan platform."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from pylutron import Lutron, LutronEntity, Output
+
+from homeassistant.components.fan import FanEntity, FanEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DOMAIN, LutronData
+from .entity import LutronDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Lutron fan platform.
+
+    Adds fan controls from the Main Repeater associated with the config_entry as
+    fan entities.
+    """
+    entry_data: LutronData = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities(
+        [
+            LutronFan(area_name, device, entry_data.client)
+            for area_name, device in entry_data.fans
+        ],
+        True,
+    )
+
+
+class LutronFan(LutronDevice, FanEntity):
+    """Representation of a Lutron fan."""
+
+    _attr_preset_mode = None
+    _attr_preset_modes = None
+    _attr_should_poll = False
+    _attr_speed_count = 3
+    _attr_supported_features = FanEntityFeature.SET_SPEED
+    _lutron_device: Output
+
+    def __init__(
+        self, area_name: str, lutron_device: LutronEntity, controller: Lutron
+    ) -> None:
+        """Initialize the fan."""
+
+        super().__init__(area_name, lutron_device, controller)
+        self._attr_extra_state_attributes = {
+            "lutron_integration_id": self._lutron_device.id
+        }
+        self._prev_percentage: int | None = None
+        self._percentage: int | None = None
+        self._oscillating: bool | None = None
+        self._direction: str | None = None
```

Comment:
I don't think we use all of them. Otherwise you can set them outside of the constructor